test/test_download.py

```
@@ -0,0 +1,92 @@
+    assert result == "temp_castro.mp3"
+    mock_audio_resp.raise_for_status.assert_called_once()
+    mock_path_open.assert_called_once_with("wb")
+
+def test_download_castro_missing_source_tag(mocker):
+    """Test download_castro raises ValueError when <source> tag is missing."""
+    mock_page_resp = mocker.MagicMock()
+    mock_page_resp.content = b"<html><body>No source here</body></html>"
+    mocker.patch("download.requests.get", return_value=mock_page_resp)
+    mocker.patch("download.requests.utils.requote_uri", side_effect=lambda x: x)
+
+    with pytest.raises(ValueError, match="Audio source tag not found in Castro page."):
```

Ruff (0.15.5)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)